test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -159,12 +155,12 @@ public void failoverTest() throws InterruptedException, ExecutionException {
             controller2.abortTransaction(stream1, txnSegments.getTxnId()).join();
         }
 
-        // Scale operation should now complete on the second controller instance.
         // Note: if scale does not complete within desired time, test will timeout. 
-        while (!scaleStatus) {
-            scaleStatus = controller2.checkScaleStatus(stream1, 0).join();
-            Thread.sleep(30000);
-        }
+        AtomicBoolean scaleDone = new AtomicBoolean(false);
+        Futures.loop(() -> !scaleDone.get(), 
```

Comment:
This is no different than a simple `while` busy-wait-loop. You can just call `join` on `controller2.checkScaleStatus` and decide from there. You'll be blocking this thread regardless, but the code will look a lot simpler.

Reply:
I know. But we have had comments asking not to use Thread.sleep in the past. So have been using this workaround in tests where we want to check something in a loop with sleeps.